src/ptf/testutils.py

```diff
@@ -1406,4 +1540,104 @@ def verify_packets_any(test, pkt, ports=[], device_number=0):
 
     test.assertTrue(received == True, "Did not receive pkt on any of ports %r for device %d" % (ports, device_number))
 
+def verify_any_packet_any_port(test, pkts=[], ports=[], device_number=0):
+    """
+    Check that _any_ of the packet is received on _any_ of the specified ports belonging to
+    the given device (default device_number is 0).
+
+    Also verifies that the packet is ot received on any other ports for this
+    device, and that no other packets are received on the device (unless --relax
+    is in effect).
+    """
+    received = False
+    index = 0
+    match_index = 0
+    (rcv_device, rcv_port, rcv_pkt, pkt_time) = test.dataplane.poll(
+        timeout=1,
+        filters=FILTERS
+    )
+    for port in ports:
+        if rcv_device != device_number:
+            continue
+        if port in ports:
+            for pkt in pkts:
+                logging.debug("Checking for pkt on device %d, port %d", device_number, port)
+                if str(rcv_pkt) == str(pkt) and rcv_port == port:
+                    received = True
+                    match_index = index
+        index += 1
+    verify_no_other_packets(test)
+
+    test.assertTrue(received == True, "Did not receive pkt on any of ports %r for device %d" % (ports, device_number))
+    return match_index
+
+def verify_packets_on_ports(test, pkts=[], ports=[], device_number=0):
```

Comment:
this is not a very good name for this function, what about verify_each_packet_on_each_port?